### PR TITLE
Automated cherry pick of #101063: tests: Spawn poststart / prestop pods on the same node as the

### LINKED ISSUE #101062: HTTP Lifecycle hook tests flake on multinode clusters
#### Which jobs are flaking:

k8s-e2e-ltsc2019-containerd-flannel-sdnoverlay-master

#### Which test(s) are flaking:

[k8s.io] Container Lifecycle Hook when create a pod with lifecycle hook should execute prestart http hook properly [NodeConformance] [Conformance]
[k8s.io] Container Lifecycle Hook when create a pod with lifecycle hook should execute poststart http hook properly [NodeConformance] [Conformance]

#### Testgrid link:

https://testgrid.k8s.io/sig-windows-networking#sac1909-containerd-flannel-sdnoverlay-stable

Note: the tests have been excluded until they are fixed.

#### Reason for failure:

In the case of multinode clusters, the http server pod and the test cluster can spawn on different nodes, which can be problematic for  poststart / prestop hooks, as they are executed by the kubelet itself, and the cross-node lifecycle hook might
fail (according to the [Kubernetes network model], it is not mandatory for kubelet to be able to access pods on a different node).

However, that is not always the case, different Networking setups allow cross-node communication between kubelet and pods, which means that the above tests are passing properly (for example: https://testgrid.k8s.io/sig-windows-networking#sac1909-containerd-flannel-sdnbridge-stable), which begs the question if this behaviour should be universal or not, especially since the tests are marked as ``Conformance``, which would imply that it should work in all cases.

[Kubernetes network model]: https://kubernetes.io/docs/concepts/cluster-administration/networking/#the-kubernetes-network-model

#### Anything else we need to know:
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

Slack: https://kubernetes.slack.com/archives/C09QYUH5W/p1617965793211800

/sig testing
/sig networking
/sig windows

### PR DESCRIPTION
**Pull request by claudiubelu**
Cherry pick of #101063 on release-1.22.

#101063: tests: Spawn poststart / prestop pods on the same node as the

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

/kind flake
/sig testing
/sig networking

/priority important-soon

#### What this PR does / why we need it:

In the case of multinode clusters, the http server pod and the test cluster can spawn on different nodes, which can be problematic for  poststart / prestop hooks, as they are executed by the kubelet itself, and the cross-node lifecycle hook might
fail (according to the [Kubernetes network model], it is not mandatory for kubelet to be able to access pods on a different node).

This PR ensures that the test pod spawns on the same node as the http server pod.

[Kubernetes network model]: https://kubernetes.io/docs/concepts/cluster-administration/networking/#the-kubernetes-network-model

#### Which issue(s) this PR fixes:

Fixes #101062

```release-note
NONE
```

### CODE REVIEW

**Comment by ehashman** [comment]
It's not permitted to change conformance tests retroactively, and generally test changes are not eligible for backport unless addressing flakes in older versions. Is this affecting 1.22 CI signal?

**Comment by claudiubelu** [comment]
> 
> 
> It's not permitted to change conformance tests retroactively, and generally test changes are not eligible for backport unless addressing flakes in older versions. Is this affecting 1.22 CI signal?

As the original issue says, there are certain scenarios for which these tests flake because the kubelet on node A cannot reach the pod on node B in order to execute these prestop / poststart hooks (according to the Kubernetes network model, it is not mandatory for kubelet to be able to access pods on a different node). This currently only affects Windows nodes with overlay networks, in particular:

https://testgrid.k8s.io/sig-windows-networking#ltsc2019-containerd-flannel-sdnoverlay-stable
https://testgrid.k8s.io/sig-windows-networking#sac2004-containerd-flannel-sdnoverlay-stable
https://testgrid.k8s.io/sig-windows-networking#ltsc2022-containerd-flannel-sdnoverlay-stable

These tests are currently excluded due to this flakiness.

tl;dr, it helps only a few scenarios, not all of them.
